### PR TITLE
Set up frontend of registration page

### PR DESCRIPTION
**Pull request by Dinferno-lol**
Set up frontend form for registration page and related tests.
This PR is related to #44

### CODE REVIEW

File: App/src/services/auth.ts [hunk 1]
```diff
@@ -25,7 +41,7 @@ export function login(dto: LoginDTO): Promise<AuthenticationResponse> {
     return axios.post("/Authentication/Login", dto);
 }
 
-export function register(dto: RegisterDTO): Promise<AuthenticationResponse> {
+export function registeration(dto: RegisterDTO): Promise<AuthenticationResponse> {
```
Comment (Marc-Eid) [inline App/src/services/auth.ts:44]: why did you change the name here?

File: App/src/services/auth.ts [hunk 2]
```diff
@@ -8,6 +9,21 @@ export interface AuthenticationResponse {
     };
 }
 
+export interface DecodedToken {
```
Comment (Marc-Eid) [inline App/src/services/auth.ts:12]: this is a duplicate interface. it already exists in services/user.ts

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (Marc-Eid) [inline .vscode/launch.json:1]: why did you add this file?

File: App/src/services/auth.ts (hunk 1 above)
Comment (Dinferno-lol) [reply App/src/services/auth.ts:44]: Another constant shared the name, I changed it so there wasn't a conflict because of a function and name having the same name, and the constant defined was used more than this function so this was changed

File: .vscode/launch.json (hunk 3 above)
Comment (Dinferno-lol) [reply .vscode/launch.json:1]: I think this got added automatically when trying to run the code via vs code, I can remove it

File: App/src/services/auth.ts (hunk 1 above)
Comment (Marc-Eid) [reply App/src/services/auth.ts:34]: I would say name it something like RegisterUser() or RegisterNewUser(). just because methods/functions do something so they are usually verbs.

Comment (JamesPartsafas) [reply App/src/services/auth.ts]: I agree with Marc that `RegisterNewUser` (also registeration is a typo).

File: App/src/services/auth.ts (hunk 2 above)
Comment (JamesPartsafas) [reply App/src/services/auth.ts]: It's not duplicated. services/user.ts contains a User interface. The DecodedToken interface was moved from the Login page to services/auth.ts since it's being used in multiple spots now.

**Review by Marc-Eid — APPROVED** [review]

File: App/src/services/auth.ts (hunk 2 above)
Comment (Marc-Eid) [reply App/src/services/auth.ts]: you are right
